src/components/YourCompany/OfficeAddresses.vue

```diff
@@ -282,171 +315,164 @@ export default class OfficeAddresses extends Mixins(CommonMixin, EntityFilterMix
+      this.originalOfficeAddresses.recordsOffice?.mailingAddress)
   }
 
-  /**
-   * Sets address data.
-   * @param loadInheritedFlags used to update inherited flags based on isSame checks if true
-   */
-  private initializeAddresses (loadInheritedFlags: boolean): void {
-    if (this.addresses.registeredOffice) {
-      this.mailingAddress = this.addresses.registeredOffice.mailingAddress
-      this.deliveryAddress = this.addresses.registeredOffice.deliveryAddress
-
-      if (loadInheritedFlags) {
-        this.inheritMailingAddress = this.isSame(
-          this.addresses.registeredOffice.mailingAddress,
-          this.addresses.registeredOffice.deliveryAddress
-        )
-      }
-      if (this.entityFilter(EntityTypes.BCOMP)) {
-        this.recMailingAddress = this.addresses.recordsOffice?.mailingAddress
-        this.recDeliveryAddress = this.addresses.recordsOffice?.deliveryAddress
-
-        if (loadInheritedFlags) {
-          this.inheritRegisteredAddress = (
-            this.isSame(
-              this.addresses.registeredOffice.deliveryAddress,
-              this.addresses.recordsOffice?.deliveryAddress
-            ) && this.isSame(
-              this.addresses.registeredOffice.mailingAddress,
-              this.addresses.recordsOffice?.mailingAddress
-            )
-          )
+  /** True if records delivery address has changed. */
+  private get recDeliveryChanged (): boolean {
+    return !this.isSame(this.getOfficeAddresses.recordsOffice?.deliveryAddress,
+      this.originalOfficeAddresses.recordsOffice?.deliveryAddress)
+  }
 
-          this.inheritRecMailingAddress = this.isSame(
-            this.addresses.recordsOffice?.mailingAddress,
-            this.addresses.recordsOffice?.deliveryAddress
-          )
-        }
-      }
-    }
+  /** True if any office address has changed. */
+  private get officeAddressesChanged (): boolean {
+    return (this.mailingChanged || this.deliveryChanged || this.recMailingChanged || this.recDeliveryChanged)
   }
 
-  //
-  // Getters (Computed Values)
-  //
-  /** Whether the address form is valid. */
+  /** True if the address form is valid. */
   private get formValid (): boolean {
     const registeredOfficeValid = this.mailingAddressValid &&
       (this.deliveryAddressValid || this.inheritMailingAddress)
 
     const recordsOfficeValid = this.inheritRegisteredAddress ||
       (this.recMailingAddressValid && (this.inheritRecMailingAddress || this.recDeliveryAddressValid))
 
-    return registeredOfficeValid && recordsOfficeValid
+    return (registeredOfficeValid && recordsOfficeValid)
   }
 
-  //
-  // Event Handlers
-  //
-  /** Sets the Registered Delivery Address to the Registered Mailing Address. */
+  /**
+   * Sets local address data and "inherited" flags from store.
+   */
+  private setLocalProperties (): void {
+    if (this.getOfficeAddresses.registeredOffice) {
+      this.mailingAddress = { ...this.getOfficeAddresses.registeredOffice.mailingAddress }
```

Comment:
Shallow copy is OK here because address object doesn't have sub-objects.

Reply:
those sub-objects in ShareStructure were the root cause of many a headaches lately.

Reply:
Yep :/